### DISCUSSION TITLE
Finding intersections in a rectangle?

### DISCUSSION THREAD

**Discussion by dov** [vsg-dev/VulkanSceneGraph#1270] [General]
In my application, I'd like the option of selecting all objects that intersect a rectangle defined by the mouse. Is there any rectangle/scene intersection support in VSG? If not, I thought that it may be implemented by creating a "label buffer" in which each pixel represents the index of the mesh in the scene, and then simply finding what labels exist in the selection. But how would I create the label image? Should I create a separate "graphics pipeline" that contain label image shaders? Can I share the vertexDraw nodes between the different pipelines. Is there an example showing how this is done?

**Comment by robertosfield** [comment]
A polytope intersector is way to tackle this type of requirement. The VSG
does yet have one but will soon.
